src/parser/jobs/mch/modules/Weaving.ts

```diff
@@ -0,0 +1,21 @@
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import CoreWeaving, {WeaveInfo} from 'parser/core/modules/Weaving'
+
+const HYPERCHARGE_ACTION_IDS = [
+	ACTIONS.AUTO_CROSSBOW.id,
+	ACTIONS.HEAT_BLAST.id,
+]
```

Comment:
This doesn't respect potential patch layering. The more forward-compatible way of defining this stuff is to define it with root keys:

```suggestion
const HYPERCHARGE_ACTIONS: (keyof ActionRoot)[] = [
	'AUTO_CROSSBOW',
	'HEAT_BLAST',
]
```

which you can then map to the IDs at parse-time when patch info is available

```ts
// (you'll probably want to do this once in init or similar)
const theIds = HYPERCHARGE_ACTIONS.map(key => this.data.actions[key].id)
```

There's some good examples of this in AST

Reply:
👍 added it for DNC too